Limelight/Stream/VideoDecoderRenderer.m

```diff
@@ -110,36 +124,104 @@ - (void)start
 
 // TODO: Refactor this
 int DrSubmitDecodeUnit(PDECODE_UNIT decodeUnit);
+void renderThreadProc(void*);
 
-- (void)displayLinkCallback:(CADisplayLink *)sender
-{
+void renderThreadProc(void* context) {
     VIDEO_FRAME_HANDLE handle;
     PDECODE_UNIT du;
+
+    VideoDecoderRenderer * renderer = (__bridge VideoDecoderRenderer *) context;
+    while(!PltIsThreadInterrupted(&renderer->_renderThread)){
+        if (LiWaitForNextVideoFrame(&handle, &du)){
+            LiCompleteVideoFrame(handle, DrSubmitDecodeUnit(du));
```

Comment:
This thread isn't necessary. It's basically just the decoder thread that moonlight-common-c would spawn on our behalf if we've didn't opt to use the pull render API.